### PR TITLE
Optimize SyncPeerStatus lock range

### PR DESCRIPTION
**Pull request by chuwen95**

### CODE REVIEW

**Review by kyonRay — APPROVED** [review]

**Review by kyonRay — APPROVED** [review]

**Review by JimmyShi22 — APPROVED** [review]

File: bcos-sync/bcos-sync/state/SyncPeerStatus.cpp [hunk 1]
```diff
@@ -117,26 +116,40 @@ bool SyncPeerStatus::updatePeerStatus(
...
         return false;
     }
-    // update the existed peer status
-    if (m_peersStatus.contains(_peer))
+
+    PeerStatus::Ptr peerStatus{nullptr};
+    {
+        ReadGuard lock(x_peersStatus);
+        // update the existed peer status
+        if (m_peersStatus.contains(_peer))
```
Comment (bxq2011hust) [inline bcos-sync/bcos-sync/state/SyncPeerStatus.cpp:124]: use find instead of contains, contains and [] will find twice

**Review by bxq2011hust — CHANGES_REQUESTED** [review]

File: bcos-sync/bcos-sync/state/SyncPeerStatus.cpp [hunk 2]
```diff
@@ -186,15 +201,23 @@ void SyncPeerStatus::foreachPeerRandom(std::function<bool(PeerStatus::Ptr)> cons
         swap(nodeIds[i], nodeIds[select]);
     }
 
-    // access _f() according to the random list
-    for (const auto& nodeId : nodeIds)
+    std::vector<PeerStatus::Ptr> peersStatusList;
     {
-        auto const& peer = m_peersStatus.find(nodeId);
-        if (peer == m_peersStatus.end())
+        ReadGuard lock(x_peersStatus);
```
Comment (bxq2011hust) [inline bcos-sync/bcos-sync/state/SyncPeerStatus.cpp:206]: if peerStatus will not be frequent updated, use std::mutex will have better performance

Comment (chuwen95) [reply bcos-sync/bcos-sync/state/SyncPeerStatus.cpp:206]: > if peerStatus will not be frequent updated, use std::mutex will have better performance

每200ms会有一个节点状态包，应该还是比较频繁的，这个先不改？

Comment (kyonRay) [reply bcos-sync/bcos-sync/state/SyncPeerStatus.cpp:207]: 下个PR我会修改peerStatus的的广播逻辑，降低频率。这个地方我下个PR再修改。

**Review by bxq2011hust — APPROVED** [review]

**Pull request merged**